hms-app/src/components/admin-tools/UserDetails.tsx

```diff
@@ -0,0 +1,220 @@
+  },
+  section: {
+    borderBottom: `1px solid ${
+      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
+    }`,
+    paddingLeft: theme.spacing.md,
+    paddingRight: theme.spacing.md,
+    paddingBottom: theme.spacing.md,
+    fontSize: theme.fontSizes.md,
+    fontWeight: 500,
+  },
+  label: {
+    textTransform: 'uppercase',
+    fontSize: theme.fontSizes.xs,
+    fontWeight: 700,
+  },
+}))
+
+export default function UserDetails(props: { userID: string }) {
+  const theme = useMantineTheme()
+  const { classes } = useStyles()
+  const { userID } = props
+  const [deleteModalOpened, setDeleteModalOpened] = useState(false)
+  const [editModalOpened, setEditModalOpened] = useState(false)
+  const [isUserError, setIsUserError] = useState(false)
+  const [isUserLoading, setIsUserLoading] = useState(true)
+  const [user, setUser] = useState({
+    id: 'string',
+    lastName: 'string',
+    firstName: 'string',
+    emailAddress: 'string',
+    roles: [],
+    skills: [],
+    imageUrl: 'string',
+    creationDate: new Date(),
+  } as User)
+
+  const loadSelectedUser = () => {
+    getUserDetails(userID).then(
+      (data) => {
+        setUser({
+          id: data.id,
+          lastName: data.lastName,
+          firstName: data.firstName,
+          emailAddress: data.emailAddress,
+          roles: data.roles,
+          skills: data.skills,
+          imageUrl: data.imageUrl,
+          creationDate: data.creationDate,
+        })
+        setIsUserLoading(false)
+        setIsUserError(false)
+      },
+      () => {
+        setIsUserLoading(false)
+        setIsUserError(true)
+      }
+    )
+  }
+
+  const deleteSelectedUser = () => {
+    deleteUser(userID).then((data) => {
+      setDeleteModalOpened(false)
+    })
+  }
+
+  useEffect(() => {
+    loadSelectedUser()
+    setIsUserLoading(true)
+  }, [userID])
+
+  const deleteModal = (
+    <Modal
+      centered
+      opened={deleteModalOpened}
+      onClose={() => setDeleteModalOpened(false)}
+      withCloseButton={false}
+    >
+      Are you sure you want to delete this user?
+      <h4>
+        Name: {user.firstName} {user.lastName}
+      </h4>
+      <h4>E-mail: {user.emailAddress}</h4>
+      <Button color={'red'} onClick={() => deleteSelectedUser()}>
+        Yes delete this user
+      </Button>
+      <p>
+        (This window will automatically closed as soon as the hackathon is
+        deleted)
+      </p>
+    </Modal>
+  )
+
+  const editModal = (
+    <Modal
+      centered
+      opened={editModalOpened}
+      onClose={() => setEditModalOpened(false)}
+      withCloseButton={false}
+    >
+      Edit User
+      <UserForm contextID={userID} />
+      {isUserLoading && <div>Loading...</div>}
+      <p>
+        (This window will automatically close as soon as the category is
+        deleted)
+      </p>
+    </Modal>
+  )
+
+  return (
+    <>
+      {isUserError && (
```

Comment:
this maybe should be `isUserError && !isUserLoading && (` then it will be hidden as soon as you try to reload :thinking: the same thing would apply to all the other components where we have `isLoading` and `isError`